lib/ansible/modules/windows/win_data_deduplication.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_data_deduplication
+version_added: "2.9"
+short_description: Module to enable Data Deduplication on a volume.
+description:
+- This module can be used to enable Data Deduplication on a Windows volume.
+- You have to have the FS-Data-Deduplication feature installed.
+- This module doesn't support check_mode due to a lack of WhatIf support.
+options:
+  drive_letter:
+    description:
+    - Windows drive letter on which to enable data deduplication.
+    required: yes
+    type: str
+  state:
+    description:
+    - Wether to enable or disable data deduplication on the selected volume.
+    required: yes
+    default: enabled
+    type: str
+    choices: [ enabled, disabled ]
+  settings:
+    description:
+    - List of settings to pass to the Set-DedupVolume powershell command.
+    - Please see the Microsoft Powershell for Windows documentation for all the available options.
+    - MinimumFileSize will default to 32768 if not defined or if the value is less than 32768.
+    required: no
+    type: list
+  dedup_job:
```

Comment:
Same thing here, having an option that just passes values into the cmdlet is a glorified API wrapper which is not what Ansible tries to do. Honestly I don't think this should be part of the module, it's not idempotent and someone can easily do the same thing with

```
- win_data_deduplication:
    ..
  register: data_dedup

- win_shell: Start-DedupJob -Type ...
  when: data_debug is changed
```